fastai/core.py

```diff
@@ -3,37 +3,36 @@
 
 def sum_geom(a,r,n): return a*n if r==1 else math.ceil(a*(1-r**n)/(1-r))
 
+def is_listy(x): return isinstance(x, (list,tuple))
+def is_iter(x): return isinstance(x, collections.Iterable)
+def map_over(x, f): return [f(o) for o in x] if is_listy(x) else f(x)
+
 conv_dict = {np.dtype('int8'): torch.LongTensor, np.dtype('int16'): torch.LongTensor,
     np.dtype('int32'): torch.LongTensor, np.dtype('int64'): torch.LongTensor,
     np.dtype('float32'): torch.FloatTensor, np.dtype('float64'): torch.FloatTensor}
 
 def A(*a):
     return np.array(a[0]) if len(a)==1 else [np.array(o) for o in a]
 
-def T(a):
-    if torch.is_tensor(a): res = a
-    else:
-        a = np.array(np.ascontiguousarray(a))
-        if a.dtype in (np.int8, np.int16, np.int32, np.int64):
-            res = torch.LongTensor(a.astype(np.int64))
-        elif a.dtype in (np.float32, np.float64):
-            res = torch.FloatTensor(a.astype(np.float32))
-        else: raise NotImplementedError(a.dtype)
-    return to_gpu(res, async=True)
+def T_(a):
+    if torch.is_tensor(a): return a
+    a = np.array(np.ascontiguousarray(a))
+    if a.dtype in (np.int8, np.int16, np.int32, np.int64):
+        return torch.LongTensor(a.astype(np.int64))
+    if a.dtype in (np.float32, np.float64):
+        return torch.FloatTensor(a.astype(np.float32))
+    raise NotImplementedError(a.dtype)
+def T(a): return to_gpu(T_(a), async=True)
 
 def create_variable(x, volatile, requires_grad=False):
     if not isinstance(x, Variable):
         x = Variable(T(x), volatile=volatile, requires_grad=requires_grad)
     return x
 
-def V_(x, requires_grad=False, volatile=False):
-    return create_variable(x, volatile=volatile, requires_grad=requires_grad)
-def V(x, requires_grad=False, volatile=False):
-    return [V_(o, requires_grad=requires_grad, volatile=volatile)
-            for o in x] if isinstance(x,(list,tuple)) else V_(x, requires_grad=requires_grad, volatile=volatile)
-
-def VV_(x): return create_variable(x, True)
-def VV(x):  return [VV_(o) for o in x] if isinstance(x,list) else VV_(x)
+def V_(x, requires_grad=False, volatile=False): return create_variable(x, volatile, requires_grad)
+def VV_(x):                                     return create_variable(x, True)
+def V(x, requires_grad=False, volatile=False): return map_over(x, lambda o: V_(o, requires_grad, volatile))
+def VV(x):                                     return map_over(x, VV_)
```

Comment:
This is a slight expansion to include `tuple` as well as `list` (which is all there was before).

Seems harmless, but I'm willing to revert.